### DISCUSSION TITLE
Is there a way to get the selected proposal via VICI - including v6.0 ML KEM

### DISCUSSION THREAD

**Discussion by JETtech-Labs** [strongswan/strongswan#2560] [Q&A]
The v6.0 ML plugin is working great in our tests and we are able to peer with selected proposals in the log displayed as IKE:AES_GCM_16_256/PRF_HMAC_SHA2_384/ECP_384/KE1_ML_KEM_1024 .

Is there a way to get the SA's selected ML KEM via VICI? When we look at the output from list-sas() we don't see any indication that ML KEM is used - just the dh_group is shown and that doesn't seem to include the ML KEM chosen.

**Comment by tobiasbrunner** [comment]
The additional key exchange methods are sent in `ake1`-`ake7` attributes.

**Reply by JETtech-Labs** [reply]
Thanks I see it now in the IKEv2 details ake1="ML_KEM_1024"
